src/index.ts

```diff
@@ -81,123 +129,158 @@ class Cents {
     this.greaterThanOrEqual = compareCentsFunction(comparators.greaterThanOrEqual, Cents);
     this.gte = this.greaterThanOrEqual;
     this.value = value;
-    if ((this.value != null ? this.value.toNumber : undefined) != null) {
+
+    if (this.value instanceof Cents || this.value instanceof BigNumber) {
       // Could be instanceof BigNumber or Cents.
       this.value = this.value.toNumber();
     }
 
     if (typeof this.value === 'string') {
       this.value = Number(this.value);
     }
+
     if (typeof this.value !== 'number') {
-      throw new Error(`${this.value} must be a Number`);
+      throw new Error(this.value);
```

Comment:
Here I was trying to avoid a eslint disable for this line since at this point, the only know about `this.value` is that is not a number, so TS complains about 
`Invalid type "never" of template literal expression.eslint[@typescript-eslint/restrict-template-expressions](https://typescript-eslint.io/rules/restrict-template-expressions)`
so after your comment I think is a good idea to disable **restrict-template-expressions** in this line instead of doing a regresion, sounds good?